crates/datasources/src/native/access.rs

```diff
@@ -85,6 +88,48 @@ impl LogStoreFactory for FakeStoreFactory {
         Ok(default_logstore(store, location, options))
     }
 }
+struct DeltaField {
+    data_type: DeltaDataType,
+    metadata: Option<HashMap<String, Value>>,
+}
+
+// Some datatypes get downgraded to a different type when they are stored in delta-lake.
+// So we add some metadata to the field to indicate that it needs to be converted back to the original type.
+fn arrow_to_delta_safe(arrow_type: &DataType) -> DeltaResult<DeltaField> {
+    match arrow_type {
+        dtype @ DataType::Timestamp(_, tz) => {
+            let delta_type =
+                (&DataType::Timestamp(TimeUnit::Microsecond, tz.clone())).try_into()?;
+            let mut metadata = HashMap::new();
+            metadata.insert("arrow_type".to_string(), json!(dtype));
```

Comment:
- what if the metadata isn't present (presumably our catalog and delta have different ideas of the types?) is this an error? should we backfill or (in theory) migrate user data?
- I don't know delta well enough, but does this meta data impact anything with delta, or impact our compatibility with delta?